operator/docs/examples/custom-authenticator.yaml

```diff
@@ -0,0 +1,31 @@
+# To create this cluster, run:
+# * export GUEST_CLUSTER_NAME="foobar"
+# * envsubst < auth-stage2.yaml | kubectl --kubeconfig $KUBECONFIG apply -f -
+
+apiVersion: kit.k8s.sh/v1alpha1
+kind: ControlPlane
+metadata:
+  name: $GUEST_CLUSTER_NAME
+spec:
+  master:
+    apiServer:
```

Comment:
I don't think this apiserver is leveraging this custom authenticator iiuc, correct ? 
Default token flag points to a different location [here](https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/operator/pkg/controllers/master/kubeapiserver.go#L142) - than what is describe in line [24](https://github.com/awslabs/kubernetes-iteration-toolkit/pull/295/files#diff-7e9b9bcfe9cfd574b984f6bbd04658af0da7c78be9b55ae4a107e37220437433R24), unless I'm missing something here.

Reply:
No it's just very confusing. There are 3 kubeconfigish flags for the authenticator:
* `--kubeconfig` -> this is configuring one direction communication from authenticator to kube-apiserver
* `--generate-kubeconfig` -> this is configuring communication from kube-apiserver to authenticator (this is the token file passed [here](https://github.com/awslabs/kubernetes-iteration-toolkit/blob/main/operator/pkg/controllers/master/kubeapiserver.go#L142).
* `--kubeconfig-pregenerated` -> boolean flag we don't use if we don't want the `--kubeconfig` flag to generate a new kubeconfig.

I'll add a comment elaborating on this in the yaml.